src/MiniProfiler.AspNetCore.Mvc/MvcExtensions.cs

```diff
@@ -23,6 +25,17 @@ public static IMiniProfilerBuilder AddMiniProfiler(this IServiceCollection servi
                     .AddTransient<IConfigureOptions<MvcViewOptions>, MiniProfilerSetup>();
             return new MiniProfilerBuilder(services);
         }
+
+        /// <summary>
+        /// Adds MiniProfiler timings for actions and views.
+        /// </summary>
+        /// <param name="services">The services collection to configure.</param>
+        /// <param name="configureOptions">An Action{MiniProfilerOptions} to configure options for MiniProfiler.</param>
+        public static IMiniProfilerBuilder AddMiniProfiler(this IServiceCollection services, Action<MiniProfilerOptions> configureOptions = null)
+        {
+            services.Configure(configureOptions);
```

Comment:
The ASPNET Classic MiniProfiler doesn't appear to depend on any explicitly defined configuration from the web.config. If the final ASPNET Core implementation is planning to do the same, and it doesn't depend on anything defined in a configuration (such as a json file). There's really no reason to use `Configure()` here. In fact you might be able to eliminate a direct dependency on `Microsoft.Extensions.Options` by using `TryAddSingleton()` or the like instead. Using Configure extension method just allows you to benefit from things offered by the Configuration like refreshing on changes, binding against a source, and being able to composite the resulting effective configuration from multiple disparate files & formats.

Reply:
I've just pushed latest which hopefully means the user very rarely has to do either, this was a mess (to the user), IMO. Hopefully latest is much cleaner all around. It's using DI on the configure for the IMemoryCache, which also means the user need not have a separate `.AddMemoryCache()` if going with app-local-in-memory defaults.